kernel.py

```diff
@@ -140,10 +140,18 @@ def move_card(self, player: Player, moving_card: Card, from_area: Area, to_area:
         :return: whether the action was performed
         """
 
-        player = self.__mutablize_obj(player)
+        requestor = self.__mutablize_obj(requestor)
+        if type(requestor) == Player:
```

Comment:
Standard practice is to use `isinstance(requestor, Player)` instead of type == due to subclassing (maybe for virtual players?) Also make sure the other branch is a correct type